src/tools/crossgen2/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -167,6 +167,12 @@ private void CompileMethodInternal(IMethodNode methodCodeNodeNeedingCode, Method
 #endif
             try
             {
+#if READYTORUN
```

Comment:
Just wondering (because there's starting to be too many `#if READYTORUN` here): If we were to keep the `ShouldSkipCompilation` call in `CompileMethod` as it was, but instead delete the `codeGotPublished` handling here and move that handling to `CompileMethod` (i.e. `CompileMethod` would have the `codeGotPublished` bool and the `finally` that would publish empty code if we didn't successfully compile things). `ShouldSkipCompiling` would be protected by that finally instead.

Would that be more readable? Feels like we could delete a bunch of `#if READYTORUN` here.

Reply:
I re-worked it so the try / finally is now in `CompileMethod`. It removes 4 `#if READYTORUN` defines